### DISCUSSION TITLE
[Feature]: Accessing loaders in router actions?

### DISCUSSION THREAD

**Discussion by aress31** [remix-run/react-router#11173] [Proposals]
### Describe what's incorrect/missing in the documentation

Is it possible to access a route `loader` in the same route `action`?

For instance, something like:

  ```js
  const userViewActionHandler = async ({ request, loader }) => {
    const { whatever } = loader;
  };
  ```

That would be super nice if this was possible from a DX perspective and definitely worth adding to the doc IMHO.

**Comment by brophdawg11** [comment]
It's not possible.  This also doesn't really make sense - loaders run after actions so the new data is not available yet.  If you want data from a prior loader execution (that's about to be made stale), you can send that along in your `formData`.

**Comment by aress31** [comment]
This is what I am doing atm, I am just passing it through the formData. But was wondering if we could improve this behaviour.

**Reply by brophdawg11** [reply]
loaders/actions in Remix are stateless API endpoints - so it's best to think of them them that way in React Router as well, even though they happen to run in a shared memory space.

That said, since you **are**  in a shared-memory space (the browser), there's nothing stopping you from sharing whatever data you want through `window` or singletons implementations.  I wouldn't recommend it, but you could.
